packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -41,38 +45,50 @@ const selectors = {
 	scheduleInput: ( attribute: string ) => `input[name="${ attribute }"]`,
 	scheduleMeridianButton: ( meridian: 'am' | 'pm' ) =>
 		`button.components-datetime__time-${ meridian }-button`,
-	scheduleResetButton: 'button.components-datetime__date-reset-button',
+	scheduleMonthSelect: `select[name="month"]`,
+
+	// Permalink
+	permalinkInput: '.components-base-control__field:has-text("URL Slug") input',
+	permalinkGeneratedURL: 'a.edit-post-post-link__link',
 
 	// Category
 	categoryCheckbox: ( categoryName: string ) =>
-		`${ sidebarParentSelector } [aria-label=Categories] :text("${ categoryName }")`,
-	tagInput: `${ sidebarParentSelector } .components-form-token-field:has-text("Add New Tag") input`,
+		`${ panel } div[aria-label=Categories] label:text("${ categoryName }")`,
 
 	// Tag
-	addedTag: ( tagName: string ) =>
-		`${ sidebarParentSelector } .components-form-token-field:has-text("Add New Tag") .components-form-token-field__token:has-text("${ tagName }")`,
-	closeSidebarButton: `${ sidebarParentSelector } [aria-label="Close settings"]:visible`, // there's a hidden copy in there
-
-	// URL Slug
-	urlSlugInput: '.components-base-control__field:has-text("URL Slug") input',
+	tagInput: `${ panel } .components-form-token-field:has-text("Add New Tag") input`,
+	addedTag: ( tag: string ) =>
+		`${ panel } .components-form-token-field__token-text:has-text("${ tag }")`,
 };
 
 /**
  * Component representing the settings sidebar in the editor.
  */
 export class EditorSettingsSidebarComponent {
-	private frame: Frame;
 	private page: Page;
+	private frameLocator: FrameLocator;
 
 	/**
 	 * Constructs an instance of the component.
 	 *
-	 * @param {Frame} frame The editor iframe to use as the page for Playwright actions.
-	 * @param page The underlying Playwright page
+	 * @param {Page} page The underlying Playwright page.
+	 * @param {FrameLocator} frameLocator Locator of the editor iframe.
 	 */
-	constructor( frame: Frame, page: Page ) {
-		this.frame = frame;
+	constructor( page: Page, frameLocator: FrameLocator ) {
 		this.page = page;
+		this.frameLocator = frameLocator;
+	}
+
+	/**
+	 * Closes the sidebar only for Mobile viewport.
+	 */
+	async closeSidebarForMobile(): Promise< void > {
+		if ( envVariables.VIEWPORT_NAME !== 'mobile' ) {
+			return;
+		}
+
+		const locator = this.frameLocator.locator( selectors.mobileCloseSidebarButton );
+		await locator.click();
```

Comment:
I've moved the method here from `GutenbergEditorPage`, since the button itself is on the `EditorSettingsSidebarComponent`.